sdks/js/packages/core/react/components/organization/plans/hooks/usePlans.tsx

```diff
@@ -155,40 +168,58 @@ export const usePlans = () => {
     [fetchActiveSubsciption]
   );
 
+  const getSubscribedPlans = useCallback(() => {
+    return subscriptions
+      .map(t => (t.plan_id ? planMap[t.plan_id] : null))
+      .filter((plan): plan is V1Beta1Plan => !!plan);
+  }, [planMap, subscriptions]);
+
+  const getTrialedPlanMaxWeightage = (plans: V1Beta1Plan[]) => {
+    return Math.max(
+      ...plans
+        .map(plan => {
+          const metadata = plan?.metadata as PlanMetadata;
+          return metadata?.weightage || 0;
+        })
+        .filter(w => w > 0)
+    );
+  };
+
   const checkAlreadyTrialed = useCallback(
     async (planIds: string[]) => {
-      try {
-        setIsTrialCheckLoading(true);
-        if (activeOrganization?.id && billingAccount?.id) {
-          const resps = await Promise.all(
-            planIds.map(planId =>
-              client?.frontierServiceHasTrialed(
-                activeOrganization?.id || '',
-                billingAccount?.id || '',
-                planId
-              )
-            )
-          );
-          const value = resps.some(resp => resp?.data?.trialed);
-          setHasAlreadyTrialed(value);
+      const subscribedPlans = getSubscribedPlans();
+      const subscribedPlansIdsSet = new Set(
+        subscribedPlans.map(plan => plan.id)
+      );
+      const maxTrialPlanWeightage = getTrialedPlanMaxWeightage(subscribedPlans);
+      const trialedPlans = allPlans.reduce((acc, plan) => {
```

Comment:
For readability, can we use map here, and pass the result to `setHasAlreadyTrialed` as follows:

```
// overall flow remains here
const trialedPlans = allPlans.map(getPlansIneligibleForTrial).filter( plan => plan)
setHasAlreadyTrialed(trialedPlans)


// details of how we are filtering plans which cannot be trialed. In future if we have any change in trial logic, only this function needs to change
function getPlansIneligibleForTrial(plan) {
        const metadata = plan?.metadata as PlanMetadata;
        const weightage = metadata?.weightage || 0;
       if (plan?.id && subscribedPlansIdsSet.has(plan?.id)) || (maxTrialPlanWeightage > 0 && weightage < maxTrialPlanWeightage) {
        return plan
    }
}
```